editor/state.js

```diff
@@ -31,9 +31,37 @@ export const blocks = combineReducers( {
 		return state;
 	},
 	order( state = [], action ) {
+		let index;
+		let swappedUid;
 		switch ( action.type ) {
 			case 'REPLACE_BLOCKS':
 				return action.blockNodes.map( ( { uid } ) => uid );
+
+			case 'MOVE_BLOCK_UP':
```

Comment:
I wonder if it would be easier to have only a single action, something like `INCREMENT_BLOCK_POSITION`, where an `increment: 1` or `increment: -1` property value could be passed to determine the increment effect.